main.go

```diff
@@ -2064,36 +2067,113 @@ func printDeviceInfo(device ios.DeviceEntry) {
 	fmt.Println(convertToJSONString(allValues))
 }
 
-func runSyslog(device ios.DeviceEntry) {
+func runSyslog(device ios.DeviceEntry, parse bool) {
 	log.Debug("Run Syslog.")
 
 	syslogConnection, err := syslog.New(device)
 	exitIfError("Syslog connection failed", err)
 
 	defer syslogConnection.Close()
 
+	var logFormatter func(string) string
+	if JSONdisabled {
+		logFormatter = rawSyslog
+	} else if parse {
+		logFormatter = parsedJsonSyslog()
+	} else {
+		logFormatter = legacyJsonSyslog()
+	}
+
 	go func() {
-		messageContainer := map[string]string{}
 		for {
 			logMessage, err := syslogConnection.ReadLogMessage()
 			if err != nil {
 				exitIfError("failed reading syslog", err)
 			}
 			logMessage = strings.TrimSuffix(logMessage, "\x00")
 			logMessage = strings.TrimSuffix(logMessage, "\x0A")
-			if JSONdisabled {
-				fmt.Println(logMessage)
-			} else {
-				messageContainer["msg"] = logMessage
-				fmt.Println(convertToJSONString(messageContainer))
-			}
+
+			fmt.Println(logFormatter(logMessage))
 		}
 	}()
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt)
 	<-c
 }
 
+func rawSyslog(log string) string {
```

Comment:
can you abye move this logic out of the main and to the syslog package? 
If someone wants to use this from the API later or just as a module having it in main.go is kind of an antipattern.
And yes, I know I am guilty of this myself :-D

Reply:
i would not move the raw and legacy parser anywhere, raw doesn't do anything and the legacy json parser is pretty much useless ;) so offering the proper parser for reuse makes sense everything else should be hidden, could even be marked as deprecated. Let me know if that makes sense

Reply:
oh yeah. Now I see it. Thanks for explaining!